hasher-matcher-actioner/terraform/authentication/main.tf

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+resource "aws_cognito_user_pool" "webapp_and_api_user_pool" {
+  account_recovery_setting {
+    recovery_mechanism {
+      name     = "verified_email"
+      priority = 1
+    }
+    recovery_mechanism {
+      name     = "verified_phone_number"
+      priority = 2
+    }
+  }
+  admin_create_user_config {
+    allow_admin_create_user_only = true
+  }
+  auto_verified_attributes = ["email"]
+  name                     = "${var.prefix}-user-pool"
+  password_policy {
+    minimum_length                   = 8
+    require_lowercase                = true
+    require_numbers                  = true
+    require_symbols                  = true
+    require_uppercase                = true
+    temporary_password_validity_days = 1
+  }
+  schema {
+    name                = "email"
+    attribute_data_type = "String"
+    mutable             = true
+    required            = true
+    string_attribute_constraints {
+      min_length = 5
+      max_length = 254
+    }
+  }
+  username_configuration {
+    case_sensitive = false
+  }
+}
+
+resource "aws_cognito_user_pool_client" "webapp_and_api_user_pool_client" {
+  name                                 = "${var.prefix}-user-pool-client"
+  user_pool_id                         = aws_cognito_user_pool.webapp_and_api_user_pool.id
+  generate_secret                      = false
+  allowed_oauth_flows_user_pool_client = true
+  explicit_auth_flows                  = ["ALLOW_CUSTOM_AUTH", "ALLOW_USER_SRP_AUTH", "ALLOW_REFRESH_TOKEN_AUTH"]
+  allowed_oauth_scopes                 = ["openid"]
+  allowed_oauth_flows                  = ["code"]
+  callback_urls                        = ["https://localhost:3000"]
```

Comment:
Just curious, how do these work? can we support them both locally and using cloudfront? Thanks!